### ISSUE TITLE
gradient can't flow in pg_losses 

### ISSUE THREAD

**Issue by zkmake520** [asyml/texar#65]
In pg_losses_with_logits function, the code 
     `actions = tf.stop_gradient(actions)`
will stop the gradient flows to the Encoder-Decoder framework which generates the actions(samples).
So in the example: seq2seq_attn_pg.py , the whole network can't get trained 
I am not sure if in your design you only want to update the Discriminator,  but the Generator will never get updated. Maybe a better implementation is to specify the different train_variables?
Correct me if I am wrong.
